src/dune_rules/simple_rules.ml

```diff
@@ -59,15 +59,25 @@ let interpret_and_add_locks ~expander locks action =
   | locks -> Action_builder.map action ~f:(Action.Full.add_locks locks)
 ;;
 
+let mark_as_unsafe_for_shared_cache action =
+  let open Action_builder.O in
+  let+ action = action in
+  (* TODO: need to set can_go_in_shared_cache to the config value *)
+  Action.Full.add_can_go_in_shared_cache false action
```

Comment:
I think once this respects your configuration option, the digests should be restored.

Reply:
They are now, indeed. Reading the config out of the workspace configuration which seems to me fairly ok.